src/commands/join.cpp

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2024 <faboussa>************************************************* */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   join.cpp                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: faboussa <faboussa@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/17 11:50:56 by faboussa          #+#    #+#             */
+/*   Updated: 2024/11/06 19:43:19 by faboussa         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <iostream>
+#include <map>
+#include <stdexcept>
+#include <string>
+
+#include "../includes/Client.hpp"
+#include "../includes/Server.hpp"
+#include "../includes/colors.hpp"
+#include "../includes/numericReplies.hpp"
+#include "../includes/serverConfig.hpp"
+#include "../includes/utils.hpp"
+
+void Server::joinChannel(const std::string &param, int fd) {
+  // Check if the user is attempting to leave all channels (using "0" as param)
+  if (isLeaveAllChannelsRequest(param)) {
+    // partAllChannels(fd); // Leave all channels function
+    return;
+  }
+
+  Client client = getClientByFd(fd);
```

Comment:
We should use pointer (or const reference) instead of creating a new Client object.

Test:
```cpp
  Client client = getClientByFd(fd);
  Client *clientPtr = &_clients.at(fd);
  const Client &clientRef = _clients.at(fd);

  std::cout << "original address: " << &_clients.at(fd) << std::endl;
  std::cout << "client address: " << &client << std::endl;
  std::cout << "clientPtr address: " << clientPtr << std::endl;
  std::cout << "clientRef address: " << &clientRef << std::endl;
```
Output:
```bash
original address: 0x4def9a8
client address: 0x1ffefff390  # new object (Client client)
clientPtr address: 0x4def9a8  # pointer (Client *clientPtr)
clientRef address: 0x4def9a8  # const reference (const Client &clientRef)


```

Reply:
ok thank you, this is interesting. i have corrected it